File: June-18/14-06-18/python/01-Game-of-strength.py
MOD = 10 ** 9 + 7
for _ in range(int(input())):
    n = int(input())
    ar = [int(x) for x in input().split()]
    ar.sort()
    maxi = ar[n-1]
    diff = 0
    val = n - 1
    
    for i in range(n-1, -1, -1):
        diff += (val * ar[i])
        val -= 2
    
    ans = diff * maxi
    ans %= MOD
    print(ans)

File: June-18/14-06-18/java/01-Game-of-strength.java

import java.util.*;

class TestClass {
    static int MOD = (int)1e9 + 7;
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        while(t-- > 0)
        {
            int n = input.nextInt();
            long ar[] = new long[n];
            for(int i = 0; i < n; i++)
                ar[i] = input.nextLong();
            
            Arrays.sort(ar);
            long maxi = ar[n-1], diff = 0, ans = 0, val = n - 1;
            
            for(int i = n - 1; i >= 0; i--)
            {
                diff += (val * ar[i]);
                val -= 2;
            }
            ans = ( (diff%MOD)* (maxi%MOD) )%MOD;
            System.out.println(ans);
        }
        
    }
}
